salt/minion.py

```diff
@@ -2425,14 +2458,21 @@ def pillar_refresh(self, force_refresh=False):
                 pillarenv=self.opts.get("pillarenv"),
             )
             try:
-                self.opts["pillar"] = yield async_pillar.compile_pillar()
+                new_pillar = yield async_pillar.compile_pillar()
             except SaltClientError:
                 # Do not exit if a pillar refresh fails.
                 log.error(
                     "Pillar data could not be refreshed. "
                     "One or more masters may be down!"
                 )
             finally:
+                current_schedule = self.opts["pillar"].get("schedule", {})
+                new_schedule = new_pillar.get("schedule", {})
+                new_pillar["schedule"] = self.pillar_schedule_refresh(
```

Comment:
If we hit the client error, `new_pillar` will throw a `NameError` right?
`else` instead of `finally`?